Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -394,90 +394,65 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
   return true;
 }
 
-// ===================================================
-/* Save the map file and save a code file */
-// ----------------
-bool PPCSymbolDB::SaveMap(const std::string& filename, bool WithCodes) const
+// Save symbol map similar to CodeWarrior's map file
+bool PPCSymbolDB::SaveSymbolMap(const std::string& filename) const
 {
-  // Format the name for the codes version
-  std::string mapFile = filename;
-  if (WithCodes)
-    mapFile = mapFile.substr(0, mapFile.find_last_of(".")) + "_code.map";
-
-  // Check size
-  const int wxYES_NO = 0x00000002 | 0x00000008;
-  if (functions.size() == 0)
+  File::IOFile f(filename, "w");
+  if (!f)
+    return false;
+
+  // Write ".text" at the top
+  fprintf(f.GetHandle(), ".text\n");
```

Comment:
Yeah, having the file handle exposed is a pretty nasty hack, any good IO API would do it's best to keep it protected. Pointer theft is not fun.